node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Nodes.java

```diff
@@ -1002,15 +1001,23 @@ private NodeMutexes lockAndGetAll(Collection<Node> nodes, Optional<Duration> tim
         }
         finally {
             // If we didn't manage to lock all nodes, we must close the ones we did lock before we throw.
-            NodeMutexes.close(locked.iterator());
+            NodeMutexes.close(locked);
         }
     }
 
     /** A node with their locks, acquired in a universal order. */
     public record NodeMutexes(List<NodeMutex> nodes) implements AutoCloseable {
-        @Override public void close() { close(nodes.iterator()); }
-        private static void close(Iterator<NodeMutex> nodes) {
-            if (nodes.hasNext()) try (NodeMutex node = nodes.next()) { close(nodes); }
+        @Override public void close() { close(nodes); }
+        private static void close(Iterable<NodeMutex> nodes) {
+            RuntimeException thrown = null;
+            for (NodeMutex node : nodes) {
```

Comment:
This changes lock release order, but that shouldn't matter. 

Actually ... come to think of it, there were some problems with instrumentation some time ago, when this was the case. I'm pretty sure that fixed, but can reverse the order here, just in case.